redbot/cogs/streams/streams.py

```diff
@@ -281,15 +282,32 @@ async def check_online(
                 _("Something went wrong whilst trying to contact the stream service's API.")
             )
         else:
+            use_buttons: bool = await self.config.guild(ctx.channel.guild).use_buttons()
             if isinstance(info, tuple):
                 embed, is_rerun = info
+                if use_buttons:
+                    stream_url = embed.url
+                    view = discord.ui.View()
+                    view.add_item(
+                        discord.ui.Button(
+                            label=_("Open Stream"), style=discord.ButtonStyle.link, url=stream_url
+                        )
+                    )
                 ignore_reruns = await self.config.guild(ctx.channel.guild).ignore_reruns()
                 if ignore_reruns and is_rerun:
                     await ctx.send(_("That user is offline."))
                     return
             else:
                 embed = info
-            await ctx.send(embed=embed)
+                if use_buttons:
+                    stream_url = embed.url
+                    view = discord.ui.View()
+                    view.add_item(
+                        discord.ui.Button(
+                            label=_("Open Stream"), style=discord.ButtonStyle.link, url=stream_url
+                        )
+                    )
+            await ctx.send(embed=embed, view=view if use_buttons else None)
```

Comment:
Code above is identical, this can be deduplicated by making this fragment not part of the `if`. To simplify this further, `view` can be set to `None` by default as then we don't have to do a ternary if inside `ctx.send()` call.
```suggestion
            view = None
            if use_buttons:
                stream_url = embed.url
                view = discord.ui.View()
                view.add_item(
                    discord.ui.Button(
                        label=_("Open Stream"), style=discord.ButtonStyle.link, url=stream_url
                    )
                )
            await ctx.send(embed=embed, view=view)
```

Reply:
From my testing I actually found that timeouts don't matter for `discord.ButtonStyle.link` style buttons, they're usable past the timeout which is why I didn't set one. You're right though that it's still better to set it explicitly.

Reply:
Hmm, you're right, we actually shouldn't set a timeout to `None` because then it forever stays in the memory. We should just keep the default so that it's at least removed from the memory after 3 minutes.

The reason why I suggested this change was that the buttons disappeared from the message after a bit which I thought was related to the timeout (but it isn't because that would require you to explicitly edit the message in `on_timeout()`)... I'm actually not sure why it happened *and if* it actually happened (maybe I confused messages?).

Would you mind removing the `timeout=` parameter?

Reply:
Done! You might've had `[p]streamset autodelete` toggled on and didn't notice the message getting deleted? Not sure